src/applications/personalization/profile/components/notification-settings/NotificationSettings.jsx

```diff
@@ -32,69 +32,74 @@ import HealthCareGroupSupportingText from './HealthCareGroupSupportingText';
 import NotificationGroup from './NotificationGroup';
 import { FieldHasBeenUpdated as FieldHasBeenUpdatedAlert } from '../alerts/FieldHasBeenUpdated';
-import { useFeatureToggle } from '~/platform/utilities/feature-toggles';
+import { MissingContactInfoExpandable } from './MissingContactInfoExpandable';
 
 const NotificationSettings = ({
-  allContactInfoOnFile,
   emailAddress,
   facilities,
   fetchNotificationSettings,
   mobilePhoneNumber,
-  noContactInfoOnFile,
-  notificationGroups,
-  shouldFetchNotificationSettings,
   shouldShowAPIError,
   shouldShowLoadingIndicator,
 }) => {
   const location = useLocation();
 
-  const { TOGGLE_NAMES, useToggleValue } = useFeatureToggle();
-  const showQuickSubmitGroup = useToggleValue(
-    TOGGLE_NAMES.profileShowQuickSubmitNotificationSetting,
-  );
-
-  React.useEffect(() => {
-    // issue: 48011
-    // used via passed state from contact info - mobile update alert link
-    if (location.state?.scrollToTop) {
-      scroll.scrollToTop({ duration: 0, smooth: false });
-    }
+  const {
+    toggles: notificationToggles,
+    useAvailableGroups,
+  } = useNotificationSettingsUtils();
 
-    focusElement('[data-focus-target]');
-    document.title = `Notification Settings | Veterans Affairs`;
-  }, []);
-
-  React.useEffect(
+  const requiredContactInfoOnFile = useMemo(
     () => {
-      if (shouldFetchNotificationSettings) {
-        fetchNotificationSettings({ facilities });
-      }
+      return notificationToggles?.showEmailNotificationSettings
+        ? !!(emailAddress || mobilePhoneNumber)
+        : !!mobilePhoneNumber;
     },
-    [fetchNotificationSettings, shouldFetchNotificationSettings],
+    [emailAddress, mobilePhoneNumber, notificationToggles],
   );
 
-  // if either phone number or email address is not set
   const showMissingContactInfoAlert = React.useMemo(
     () =>
       !shouldShowLoadingIndicator &&
       !shouldShowAPIError &&
-      !allContactInfoOnFile,
-    [allContactInfoOnFile, shouldShowAPIError, shouldShowLoadingIndicator],
+      !requiredContactInfoOnFile,
+    [requiredContactInfoOnFile, shouldShowAPIError, shouldShowLoadingIndicator],
+  );
+
+  const shouldFetchNotificationSettings = useMemo(
+    () => {
+      return !showMissingContactInfoAlert && !shouldShowAPIError;
+    },
+    [showMissingContactInfoAlert, shouldShowAPIError],
+  );
+
+  React.useEffect(
+    () => {
+      // issue: 48011
+      // used via passed state from contact info - mobile update alert link
+      if (location.state?.scrollToTop) {
+        scroll.scrollToTop({ duration: 0, smooth: false });
+      }
+
+      focusElement('[data-focus-target]');
+      document.title = `Notification Settings | Veterans Affairs`;
+    },
+    [location.state?.scrollToTop],
   );
 
-  // shown as long as we aren't loading data and they have at least one
-  // communication channel on file
-  const showNotificationOptions = React.useMemo(
+  React.useEffect(
```

Comment:
similar to above: would importing` { useEffect } from 'react'` and replacing `React.useEffect()` with `useEffect()` behave the same?

Reply:
yea this is a good call. These were left over old code, so I'm totally on board with cleaning em up for consistency.